cpp/src/round/round.cu

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column_factories.hpp>
+#include <cudf/copying.hpp>
+#include <cudf/detail/nvtx/ranges.hpp>
+#include <cudf/detail/round.hpp>
+#include <cudf/round.hpp>
+#include <cudf/scalar/scalar.hpp>
+#include <cudf/types.hpp>
+#include <cudf/utilities/type_dispatcher.hpp>
+#include <type_traits>
+
+namespace cudf {
+
+namespace detail {
+
+float __device__ generic_round(float f) { return roundf(f); }
+double __device__ generic_round(double d) { return ::round(d); }
+
+float __device__ generic_modf(float a, float* b) { return modff(a, b); }
+double __device__ generic_modf(double a, double* b) { return modf(a, b); }
+
+template <typename T, typename std::enable_if_t<std::is_signed<T>::value>* = nullptr>
+T __device__ generic_abs(T value)
+{
+  return abs(value);
+}
+
+template <typename T, typename std::enable_if_t<not std::is_signed<T>::value>* = nullptr>
+T __device__ generic_abs(T value)
+{
+  return value;
+}
+
+template <typename T, typename std::enable_if_t<std::is_signed<T>::value>* = nullptr>
+bool __device__ is_negative(T value)
+{
+  return value >= 0;
+}
+
+// this is needed to suppress warning: pointless comparison of unsigned integer with zero
+template <typename T, typename std::enable_if_t<not std::is_signed<T>::value>* = nullptr>
+bool __device__ is_negative(T value)
```

Comment:
This logic looks like these functions should be called `is_positive`.

Reply:
Wow, thank you so much for catching this. I actually originally had it called `is_positive` but technically that isn't true for `>= 0` so I wanted to flip it, but apparently I got distracted and only did a rename: https://github.com/rapidsai/cudf/pull/6562/commits/22ac51c5f9c18d67bf822d0793cb81d639838d4c

Reply:
Fixed.

Reply:
`is_negative` still returns `true` for unsigned types. That doesn't seem right.

Reply:
:man_facepalming:

Reply:
Fixed, plus a missing unit test that should have been there to catch this mistake.